frontend/src/components/accountPage/UploadProgressModal/EditableTable.tsx

```diff
@@ -0,0 +1,210 @@
+/* eslint-disable react/no-multi-comp */
+import { Table, Input, Form, Popconfirm } from 'antd';
+import {
+  EditOutlined,
+  CheckOutlined,
+  CloseOutlined,
+  DeleteOutlined,
+} from '@ant-design/icons';
+import { UserAccountPage } from '../../../utils';
+import { FC, useState } from 'react';
+interface EditableCellProps {
+  editing: boolean;
+  dataIndex: string;
+  title: any;
+  inputType: 'number' | 'text';
+  record: UserAccountPage;
+  index: number;
+  children: React.ReactNode;
+}
+
+export interface IEditableTableProps {
+  data: UserAccountPage[];
+  updateUserCSV: (user: UserAccountPage[]) => void;
+  setEditing: (value: boolean) => void;
+}
+
+const EditableTable: FC<IEditableTableProps> = props => {
+  const [form] = Form.useForm();
+  const [editingKey, setEditingKey] = useState<string>('');
+
+  const isEditing = (record: UserAccountPage) => record.key === editingKey;
+
+  const edit = (record: Partial<UserAccountPage> & { key: React.Key }) => {
+    form.setFieldsValue({
+      userid: '',
+      name: '',
+      surname: '',
+      email: '',
+      ...record,
+    });
+    setEditingKey(record.key);
+  };
+
+  const cancel = () => {
+    setEditingKey('');
+    props.setEditing(false);
+  };
+  const EditableCell: React.FC<EditableCellProps> = ({
+    editing,
+    dataIndex,
+    title,
+    inputType,
+    record,
+    index,
+    children,
+    ...restProps
+  }) => {
+    const inputNode = <Input />;
+
+    return (
+      <td {...restProps}>
+        {editing ? (
+          <Form.Item
+            name={dataIndex}
+            style={{ margin: 0 }}
+            rules={[
+              {
+                required: true,
+                message: `Please Input ${title}!`,
+              },
+            ]}
+          >
+            {inputNode}
+          </Form.Item>
+        ) : (
+          children
+        )}
+      </td>
+    );
+  };
```

Comment:
Would put cell related things in another file (in this same folder) to keep it more organized